admin/ownership_grants/V1.3.1__synapse_data_warehouse_prod.sql

```diff
@@ -0,0 +1,93 @@
+---- RBAC reconfiguration of data warehouse ----
+-- The following grants transfer ownership of current and future 
+-- data warehouse objects from SYSADMIN to each namespace's respective
+-- `*ALL_ADMIN` database role.
+
+---- SYNAPSE_DATA_WAREHOUSE ----
+GRANT OWNERSHIP
+    ON DATABASE SYNAPSE_DATA_WAREHOUSE
+    TO ROLE SYNAPSE_DATA_WAREHOUSE_ADMIN
+	COPY CURRENT GRANTS;
+
+-- SYNAPSE
+GRANT OWNERSHIP
+	ON SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP
+	ON ALL TABLES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+-- GRANT OWNERSHIP
+-- 	ON ALL DYNAMIC TABLES
+-- 	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+-- 	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+-- 	COPY CURRENT GRANTS;
+GRANT OWNERSHIP
+	ON ALL DYNAMIC TABLES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO ROLE SYSADMIN 
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP
+	ON ALL STAGES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP
+	ON ALL VIEWS
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+	COPY CURRENT GRANTS;
+-- GRANT OWNERSHIP
+-- 	ON ALL TASKS
+-- 	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+-- 	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_ALL_ADMIN
+-- 	COPY CURRENT GRANTS;
+GRANT OWNERSHIP
+	ON ALL TASKS
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE
+	TO ROLE SYSADMIN
+	COPY CURRENT GRANTS;
+
+-- SYNAPSE_RAW
+GRANT OWNERSHIP
+	ON SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP
+	ON ALL TABLES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP
+	ON ALL STAGES
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW_ALL_ADMIN
+	COPY CURRENT GRANTS;
+GRANT OWNERSHIP
+	ON ALL STREAMS
+	IN SCHEMA SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW
+	TO DATABASE ROLE SYNAPSE_DATA_WAREHOUSE.SYNAPSE_RAW_ALL_ADMIN
```

Comment:
Would merging this be premature since we were discussing potentially granting ownership of everything to a separate database role?

Reply:
On streams, specifically? I don't think so. Streams are created on tables which exist in just one schema, so streams should be granted to the appropriate `*ALL_ADMIN` role.

The only objects which I foresee being affected by the multi-schema privilege work are the objects which we had to hand back to SYSADMIN recently (dynamic tables and tasks).

Reply:
Got it! Thanks! Let me take another look through this then